src/System.IO.Pipelines/MemoryPool.cs

```diff
@@ -75,7 +75,6 @@ public override OwnedBuffer<byte> Rent(int size)
             }
 
             var block = Lease();
-            block.Initialize();
```

Comment:
I am not super fimiliar with MemoryPoolBlock and MemoryPoolSlab. Since OwnedBuffer does not have a T array field, what should initialize do exactly?